### PR TITLE
Do not evaluate vertex/edge labels unless constraint checking is enabled

### LINKED ISSUE #1081: Too early labels evaluation for constraint checking is causing insertion slowdown
Commit https://github.com/JanusGraph/janusgraph/commit/f7ee75710ed7a1f22ce5ab657a9d4eaf2bd855a3 introduces constraints checking, but unfortunately code in StandardJanusGraphTx.java evaluates labels for constraint checking before verification that constraint are enabled, which may lead to noticeable performance drop when adding edges (probably vertices too) - https://groups.google.com/forum/#!topic/janusgraph-users/KecQAyu1dUk

### PR DESCRIPTION
**Pull request by YevIgn**
Fixes #1081 

Signed-off-by: Evgeniy Ignatiev <yevgeniy.ignatyev@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by YevIgn** [comment]
Already filled and sent CLA - the processing is in progress.

**Comment by mbrukman** [comment]
CLA is addressed in https://github.com/JanusGraph/legal/pull/95

File: janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphOperationCountingTest.java [hunk 1]
```diff
@@ -236,6 +236,8 @@ public void testSettingProperty() {
         mgmt.makePropertyKey("foo").dataType(String.class).cardinality(Cardinality.SINGLE).make();
         finishSchema();
 
+        clopen(option(SCHEMA_CONSTRAINTS), true);
```
Comment (analytically) [inline janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphOperationCountingTest.java]: how is this related to label evaluation?

Comment (YevIgn) [reply janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphOperationCountingTest.java]: This test case was changed in the f7ee75710ed7a1f22ce5ab657a9d4eaf2bd855a3 with additional validations - https://github.com/JanusGraph/janusgraph/pull/923/files#diff-19c61b5c821437fcb295867d83c73a0c - that will fail after moving vertexLabel()/edgeLabel() methods under the constraints enabled check, unless checks are enabled explicitly.

Comment (farodin91) [reply janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphOperationCountingTest.java]: It would be also possible to revert changes on this test class from PR #923.

Comment (pluradj) [reply janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphOperationCountingTest.java]: I would not revert the changes on the test class unless the test scenario is no longer valid, but I don't see why that would be the case.
